src/test/java/it/smartcommunitylab/aac/saml/SamlIdentityProviderConfigurationTest.java

```diff
@@ -0,0 +1,320 @@
+            Assertions.assertThat(idpCfgMap.getRequireEmailAddress()).isEqualTo(rawCfgMap.get("requireEmailAddress"));
+        }
+    }
+
+    /**
+     * This tests checks if s saml configuration map matches the content of the given saml configuration
+     * @throws Exception
+     */
+    @Test
+    public void validateSamlIdentityProviderConfigCreation() throws Exception {
+
+        // Create (hardcoded) configuration map to be used for initialization
+        Map<String, Serializable> rawCfgMap = buildRawConfigurationMap(true);
+        SamlIdentityProviderConfigMap idProvCfgMap = new SamlIdentityProviderConfigMap();
+        idProvCfgMap.setConfiguration(rawCfgMap);
+
+        ConfigurableIdentityProvider cip = getDefaultConfigurableIdentityProvider();
+        IdentityProviderSettingsMap idProvSettingMap = getDefaultIdentityProviderSettingsMap();
+
+        SamlIdentityProviderConfig IdPCfg = new SamlIdentityProviderConfig(cip, idProvSettingMap, idProvCfgMap);
+        Assertions.assertThat(IdPCfg.getEntityId()).isEqualTo(expEntityID);
+    }
+
+    /**
+     * This test checks that a RelyingPartyRegistration contains information matching the saml configuration (with automatic idp metadata discovery)
+     * @throws Exception
+     */
+    @Test
+    public void validateRelyingPartyRegistrationAutomatic() throws Exception {
+
+        boolean automaticDiscoverySetting = true;
+        // Create (hardcoded) configuration map to be used for initialization
+        Map<String, Serializable> rawCfgMap = buildRawConfigurationMap(automaticDiscoverySetting);
+        SamlIdentityProviderConfigMap idProvCfgMap = new SamlIdentityProviderConfigMap();
+        idProvCfgMap.setConfiguration(rawCfgMap);
+
+        ConfigurableIdentityProvider cip = getDefaultConfigurableIdentityProvider();
+        IdentityProviderSettingsMap idProvSettingMap = getDefaultIdentityProviderSettingsMap();
+
+        SamlIdentityProviderConfig idProvCfg = new SamlIdentityProviderConfig(cip, idProvSettingMap, idProvCfgMap);
+        // Run mock server to locally catch the identity provider metadata request.
+        String IdpMetadata = fetchIdpMetadataExample();
+        WireMockServer mockIdpServer = buildMockIdpServer(IdpMetadata);
+        mockIdpServer.start();
+
+        // check that the obtained saml Identity Provider matches the obtained map
+        RelyingPartyRegistration rpRegistration = idProvCfg.getRelyingPartyRegistration();
+        mockIdpServer.stop();
+
+        Assertions.assertThat(rpRegistration.getEntityId()).isEqualTo(expEntityID);
+        Assertions.assertThat(rpRegistration.getRegistrationId()).isEqualTo(expProviderId);
+        Assertions.assertThat(rpRegistration.getSigningX509Credentials().size()).isEqualTo(1);
+        for (Saml2X509Credential credential : rpRegistration.getSigningX509Credentials()) {
+            // NOTE: these assertions currently assume that there is only one x509credential(s) -> for-loop is trivial
```

Comment:
the test checks against a **known** configuration. If there is *one* credential you should check both:
- the credential is there and is the same
- there are NO other credentials